pandas/core/arrays/arrow/array.py

```diff
@@ -782,9 +798,14 @@ def _evaluate_op_method(self, other, op, arrow_funcs) -> Self:
 
         pc_func = arrow_funcs[op.__name__]
         if pc_func is NotImplemented:
+            if pa.types.is_string(pa_type) or pa.types.is_large_string(pa_type):
+                raise TypeError(self._op_method_error_message(other_original, op))
             raise NotImplementedError(f"{op.__name__} not implemented.")
 
-        result = pc_func(self._pa_array, other)
+        try:
+            result = pc_func(self._pa_array, other)
+        except pa.lib.ArrowNotImplementedError as err:
+            raise TypeError(self._op_method_error_message(other_original, op)) from err
```

Comment:
I am doing `from err` here, so the original pyarrow message like _"ArrowNotImplementedError: Function 'binary_join_element_wise' has no kernel matching input types (large_string, int64, large_string)"_ is still visible up in the traceback. 

For the new string dtype, I would actually be fine with suppressing this (using `from None`) to limit the length of the error traceback. But for usage of ArrowDtype in general, I suppose this information is still very useful to see.

But as I mentioned in the top post, I could also only do this try/except in case we are in ArrowStringArray, and leave the generic ArrowExtensionArray as is (raising the pyarrow error directly)

Reply:
I like the `from err` - I don't see a huge harm in including all of that information